trie-db/src/iterator.rs

```diff
@@ -221,6 +228,38 @@ impl<'a, L: TrieLayout> TrieIterator<L> for TrieDBNodeIterator<'a, L> {
 			node_hash = next_node_hash;
 		}
 	}
+
+	/// Advance the iterator into a prefix, no value out of the prefix will be accessed
+	/// or returned after this operation.
+	pub fn prefix(&mut self, prefix: &[u8]) -> Result<(), TrieHash<L>, CError<L>> {
+		self.trail.clear();
+		self.key_nibbles.clear();
+		if self.seek_prefix(prefix)? {
+			if let Some(v) = self.trail.pop() {
+				if self.trail.len() > 0 {
```

Comment:
nit: This `if` check is unnecessary, you can just do

```rust
self.trail.clear();
self.trail.push(v);
```

which works in both cases.